File: puzzles/ts/defibrillators/defibrillators.ts
// Convert degrees to radians
function convertToRadians(angle: string): number {
  let degrees: number = Number(angle.replace(',', '.'));
  return degrees * Math.PI / 180;
}

const lon: number = convertToRadians(readline());
const lat: number = convertToRadians(readline());
const n: number = parseInt(readline());

let minDistance: number = Infinity;
let defibName: string = "";
for (let i: number = 0; i < n; i++) {
  const defib: string = readline().split(';');
  const defibLon: number = convertToRadians(defib[4]);
  const defibLat: number = convertToRadians(defib[5]);
  const x: number = (defibLon - lon) * Math.cos((lat + defibLat) / 2)
  const y: number = defibLat - lat;
  const distance: number = Math.hypot(x, y) * 6371;
  if (distance < minDistance) {
    minDistance = distance;
    defibName = defib[1];
  }
}

console.log(defibName);

File: puzzles/kotlin/src/defibrillators/defibrillators.kt
import java.util.Scanner

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val lon = toRadians(input.next())
  val lat = toRadians(input.next())
  val n = input.nextInt()
  if (input.hasNextLine()) {
    input.nextLine()
  }
  var min = Double.POSITIVE_INFINITY
  var name = ""

  for (i in 0 until n) {
    val defib = input.nextLine().split(";")
    val defibLon = toRadians(defib[4])
    val defibLat = toRadians(defib[5])
    val x = (defibLon - lon) * Math.cos((lat + defibLat) / 2)
    val y = defibLat - lat
    val distance = Math.hypot(x, y) * 6371
    if (distance < min) {
      min = distance
      name = defib[1]
    }
  }

  println(name)
}

// Convert degrees to radians
fun toRadians(angle: String): Double {
  return Math.toRadians(angle.replace(',', '.').toDouble())
}
